File: src/go/gateway-service/delivery/gql_location.go
package delivery

import "github.com/graphql-go/graphql"

var locationType = graphql.NewObject(
	graphql.ObjectConfig{
		Name: "Location",
		Fields: graphql.Fields{
			"address": &graphql.Field{
				Type: graphql.String,
			},
			"city": &graphql.Field{
				Type: graphql.String,
			},
			"zip_code": &graphql.Field{
				Type: graphql.String,
			},

			"latitude": &graphql.Field{
				Type: graphql.Float,
			},
			"longitude": &graphql.Field{
				Type: graphql.Float,
			},
		},
	},
)

File: src/jvm/gateway-service/app/ryss/gateway/gql/GQLLocation.kt
package app.ryss.gateway.gql

import graphql.schema.GraphQLObjectType

/**
 * GraphQL location type.
 */
val locationType: GraphQLObjectType = GraphQLObjectType.newObject()
    .name("Location")
    .addField("address", String.toGraphQLType())
    .addField("city", String.toGraphQLType())
    .addField("zip_code", String.toGraphQLType())
    .addField("latitude", Float.toGraphQLType())
    .addField("longitude", Float.toGraphQLType())
    .build()